### PR TITLE
TPC IDCs: Adding consistency check for received IDCs

### PR DESCRIPTION
**Pull request by matthias-kleiner**
- setting Lifetime to Sporadic due to buffering of IDCs for several TFs

### CODE REVIEW

File: Detectors/TPC/workflow/include/TPCWorkflow/TPCFLPIDCSpec.h [hunk 1]
```diff
@@ -233,7 +233,7 @@ DataProcessorSpec getTPCFLPIDCSpec(const int ilane, const std::vector<uint32_t>&
     inputSpecs.emplace_back(InputSpec{"idcs", gDataOriginTPC, TPCIntegrateIDCDevice::getDataDescription(TPCIntegrateIDCDevice::IDCFormat::Sim), subSpec, Lifetime::Timeframe});
 
     const Side side = CRU(cru).side();
-    outputSpecs.emplace_back(ConcreteDataMatcher{gDataOriginTPC, TPCFLPIDCDevice::getDataDescriptionIDCGroup(side), subSpec});
+    outputSpecs.emplace_back(ConcreteDataMatcher{gDataOriginTPC, TPCFLPIDCDevice::getDataDescriptionIDCGroup(side), subSpec}, Lifetime::Sporadic);
```
Comment (davidrohr) [inline Detectors/TPC/workflow/include/TPCWorkflow/TPCFLPIDCSpec.h:236]: Don't you provide an output here for every TF?
I assume rather the input lifetime on the EPN should be sporadic. The output lifetime can remain timeframe, as the device on the FLP will produce an output for every TF that comes in, right?
It is allowed to match lifetime output with sporadic input, just not the other way around.

Comment (matthias-kleiner) [reply Detectors/TPC/workflow/include/TPCWorkflow/TPCFLPIDCSpec.h:236]: No, on the FLP we are buffering the IDCs for 400TFs and then send them to the EPN. On the EPN the input is defined as sporadic.
Should I change it back to timeframe?

Comment (davidrohr) [reply Detectors/TPC/workflow/include/TPCWorkflow/TPCFLPIDCSpec.h:236]: OK, so this is the outputSpec on the FLP? And this means this device is not sending an output every time the run() function is called? In that case, it must be sporadic indeed.
Basically, the output must be timeframe if you send it every time run() is called. If you don't send it every time, it must be sporadic.

Comment (matthias-kleiner) [reply Detectors/TPC/workflow/include/TPCWorkflow/TPCFLPIDCSpec.h:236]: Yes, the run is called every TF, but the output is not sent every TF. Then I leave it at sporadic

**Pull request merged**